### PR TITLE
fix: upgrade algolia docsearch to v3

### PR DESCRIPTION
**Pull request by babadofar**
Magically, this also seems to make search also work a little bit better on mobile 
image

### CODE REVIEW

File: static/css/style.css [hunk 1]
```diff
@@ -62,3 +62,5 @@ a img {
         background-position: top center !important;
     } 
 }
+
+.DocSearch-Modal { top: 27px;}
```
Comment (ankitm123) [inline static/css/style.css:66]: NIT: new line

File: package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   },
   "dependencies": {
     "autoprefixer": "^9.8.5",
-    "docsearch.js": "^2.6.3",
+    "@docsearch/js": "^3.0.0-alpha.50",
```
Comment (ankitm123) [inline package.json:13]: alpha? No stable release yet?

Comment (babadofar) [reply package.json:13]: Coming soon? https://github.com/algolia/docsearch/issues/1303
"We are actually looking at open issues to see if we encounter any blockers for the stable v3 release. The next release ideally be the stable one!"

Can we get a PR env for this? I am in contact with Algolia support, they will help us set up the search, and public  access to a working environment would be very helpful

File: static/css/style.css (hunk 1 above)
Comment (babadofar) [reply static/css/style.css:66]: Where? At the end?

File: layouts/partials/scripts.html [hunk 3]
```diff
@@ -12,13 +12,14 @@
...
-    inputSelector: '#search-input',
-    algoliaOptions: { 'facetFilters': ["lang:{{ .Page.Language.Lang }}"] },
-    debug: false
-  });
+
+docsearch({
+  container: '#docsearch',
+  appId: 'GIN1OS4UKS',
+  indexName: 'jenkins-x',
+  apiKey: 'a9c99be0d65d1a8f6059be9104a12cd4',
```
Comment (ankitm123) [inline layouts/partials/scripts.html:20]: Is this a search only api-key? https://www.algolia.com/doc/guides/security/api-keys/#search-only-api-key

File: static/css/style.css (hunk 1 above)
Comment (ankitm123) [reply static/css/style.css:66]: yeah. github shows  a red circle when that's missing.

File: layouts/partials/scripts.html (hunk 3 above)
Comment (babadofar) [reply layouts/partials/scripts.html:20]: Yes it is

**Comment by ankitm123** [comment]
@babadofar need to rebase with latest main branch.

**Pull request merged**